src/domain/use-cases/change-password.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  type HashGenerator,
+  type UpdateGuardianPasswordRepository,
+  type LoadGuardianByIdRepository
+} from '@/data/protocols'
+
+export interface ChangePassword {
+  change: (userData: ChangePassword.Params) => Promise<ChangePassword.Result>
+}
+
+export namespace ChangePassword {
+  export interface Params {
+    id: string
+    password: string
+  }
+
+  export interface Result {
```

Comment:
This is the first CRUD route, will be good if other routes follow a type Result standart.